coreiface/tests/routing.go

```diff
@@ -28,7 +30,12 @@ func (tp *TestSuite) testRoutingPublishKey(t *testing.T, ctx context.Context, ap
 		t.Fatal(err)
 	}
 
-	entry, err := api.Name().Publish(ctx, p)
+	// TODO: make sure we are fine with that implicit allow offline
```

Comment:
Maybe you could add `options.NamePublishOption` in the arguments and let the caller decide. Just to be more flexible in the future.

Reply:
That make sense, fixed!